### PR TITLE
aruha-64 return validation error message

### PR DESCRIPTION
**Pull request by rcillo**
Returning a boolean is not sufficient to communicate the reason of the
validation failure, which is a system requirement.

This is the first PR in the aruha-64 feature.

I'll do it in the smallest usable chunks that I can in order to get early feedback.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/zalando/nakadi/pull/122" target="_blank">ReviewNinja

**Comment by rcillo** [comment]
I had added a new commit improving tests https://github.com/zalando/nakadi/commit/29c172fb10c9e7051235037b912985e4ec759c62

Please, I need another :star: :smile:

File: src/main/java/de/zalando/aruha/nakadi/validation/EventValidation.java [hunk 1]
```diff
@@ -52,11 +53,13 @@ public EventTypeValidator(final EventType eventType) {
...
-        // FIXME: return information on the validation place that failed
-        final boolean allItemsPass = validators.stream().anyMatch(ev -> !ev.accepts(event));
-        Preconditions.checkState(!allItemsPass, "Some validation failed");
+    public Optional<ValidationError> validate(final JSONObject event) {
+        return validators
+                .stream()
+                .map(validator -> validator.accepts(event))
+                .filter(errorOptional -> errorOptional.isPresent())
+                .findFirst()
+                .map(errorOptionalOptional -> errorOptionalOptional.get());
```
Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/validation/EventValidation.java]: can be simplified a little bit:
```
return validators
                .stream()
                .map(validator -> validator.accepts(event))
                .filter(Optional::isPresent)
                .findFirst()
                .orElse(Optional.empty());
```

File: src/main/java/de/zalando/aruha/nakadi/validation/FieldNameMustBeSet.java [hunk 2]
```diff
@@ -5,16 +5,21 @@
...
 public class FieldNameMustBeSet extends ValidationStrategy {
 
     public static final String NAME = "field-name-must-be-set";
     private final EventValidator executor = new EventValidator() {
 
         @Override
-        public boolean accepts(final JSONObject event) {
-            return event.get("name") != null;
+        public Optional<ValidationError> accepts(final JSONObject event) {
+            if (event.has("name") && event.get("name") != null) {
```
Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/validation/FieldNameMustBeSet.java]: is it a possible case that `event.has("name") == true and event.get("name") == null` ?

Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/validation/FieldNameMustBeSet.java]: I think so. The first check `event.has("name")` checks if there is such a key in the object. But the value could still be `null`.

Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/validation/FieldNameMustBeSet.java]: ok, good.

File: src/main/java/de/zalando/aruha/nakadi/validation/EventValidation.java (hunk 1 above)
Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/validation/EventValidation.java]: Nice :+1: done here https://github.com/zalando/nakadi/commit/5156215f37e5acb75fdb566d074f943f57099e46

**Pull request merged**